### PR TITLE
[php8] php warnings on case dashboard

### PR DESCRIPTION
**Pull request by demeritcowboy**
Overview
----------------------------------------
php warnings on case dashboard

Before
----------------------------------------
1. Case dashboard
2. Lots of warnings about count and url

After
----------------------------------------

Technical Details
----------------------------------------
The array has holes.

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Case/BAO/Case.php [hunk 1]
```diff
@@ -768,17 +768,19 @@ public static function getCasesSummary($allCases = TRUE) {
 
     $caseTypes = CRM_Case_PseudoConstant::caseType();
     $caseStatuses = CRM_Case_PseudoConstant::caseStatus();
-    $caseTypes = array_flip($caseTypes);
```
Comment (demeritcowboy) [inline CRM/Case/BAO/Case.php:771]: This was only used on line 845, but the flip would now be confusing.

File: tests/phpunit/CRM/Case/BAO/CaseTest.php [hunk 2]
```diff
@@ -53,7 +53,7 @@ public function testCaseActivity(): void {
     $openCaseType = CRM_Core_PseudoConstant::getKey('CRM_Activity_BAO_Activity', 'activity_type_id', 'Open Case');
     while ($res->fetch()) {
       $message = 'Failed asserting that the case activity query has a activity_type_id property:';
-      $this->assertObjectHasAttribute('activity_type_id', $res, $message . PHP_EOL . print_r($res, TRUE));
+      $this->assertObjectHasProperty('activity_type_id', $res, $message . PHP_EOL . print_r($res, TRUE));
```
Comment (demeritcowboy) [inline tests/phpunit/CRM/Case/BAO/CaseTest.php:56]: It doesn't show in the jenkins output but it gives me a warning this is deprecated, so I updated it.

Comment (demeritcowboy) [reply tests/phpunit/CRM/Case/BAO/CaseTest.php:56]: Interesting - this was changed in a point release - 9.6.11, so isn't available in 9.6.5.

**Comment by mlutfy** [comment]
Merged based on passing tests, a quick tests, and quick code review. Thanks @demeritcowboy !

**Pull request merged**